dracut/repartition/endless-repartition.sh

```diff
@@ -191,6 +196,11 @@ udevadm settle
 if [ -x /usr/sbin/amlogic-fix-spl-checksum ]; then
   /usr/sbin/amlogic-fix-spl-checksum $root_disk
   udevadm settle
+elif [ "$pt_label" = "gpt" ]; then
+  # Remove the stale boot entry pointing to the old ESP's UUID.
+  # fallback.efi will take care of creating a new one on the next boot
+  boot_num=$(efibootmgr -v -d $root_disk | grep -i $orig_esp_uuid | grep "Endless" | cut -c 5-8)
```

Comment:
If we end up using this approach, I think this will do the wrong thing on dual-boot systems. The ESP in the `endless.img` (the file mapped as $root_disk) is not used; the ESP on the host drive is used. So, this line will assign empty string to `boot_num` and then the following `efibootmgr` invocation will be wrong.

In the dual-boot case, changing the UUIDs will actually have no effect at all on the boot entry so we could just skip this part there.